### PR TITLE
Add prettier workflow

### PR DESCRIPTION
**Pull request by angrykoala**
Adds prettier 3.0.3 and adds it to CI

### CODE REVIEW

File: src/namespaces/apoc/cypher/run-first-column.ts [hunk 1]
```diff
@@ -92,12 +92,15 @@ class RunFirstColumnFunction extends CypherFunction {
...
-        }, {} as Record<string, string>);
+        const params = variables.reduce(
+            (acc, variable) => {
+                const globalScopeName = variable.getCypher(env);
+                const key = env.getReferenceId(variable);
+                acc[key] = globalScopeName;
+                return acc;
+            },
+            {} as Record<string, string>
+        );
```
Comment (mjfwebb) [inline src/namespaces/apoc/cypher/run-first-column.ts:103]:
```suggestion
        const params = variables.reduce(
            (acc: Record<string, string>, variable) => {
                const globalScopeName = variable.getCypher(env);
                const key = env.getReferenceId(variable);
                acc[key] = globalScopeName;
                return acc;
            },
            {}
        );
```

I know it's not the main point of this PR but I just wanted to point ou the "more correct" way of assigning a type in a reducer.

File: src/pattern/PatternElement.ts [hunk 2]
```diff
@@ -45,10 +45,13 @@ export abstract class PatternElement<T extends NodeRef | RelationshipRef> implem
...
-            acc[key] = param.getCypher(env);
-            return acc;
-        }, {} as Record<string, string>);
+        const paramValues = Object.entries(parameters).reduce(
+            (acc, [key, param]) => {
+                acc[key] = param.getCypher(env);
+                return acc;
+            },
+            {} as Record<string, string>
+        );
```
Comment (mjfwebb) [inline src/pattern/PatternElement.ts:54]:
```suggestion
        const paramValues = Object.entries(parameters).reduce(
            (acc: Record<string, string>, [key, param]) => {
                acc[key] = param.getCypher(env);
                return acc;
            },
            {}
        );
```

File: package.json [hunk 3]
```diff
@@ -54,6 +55,7 @@
         "jest": "29.7.0",
         "jest-extended": "4.0.1",
         "neo4j-driver": "^5.9.2",
+        "prettier": "^3.0.3",
```
Comment (mjfwebb) [inline package.json:58]: I notice some of the dependencies are pinned to a certain version and some are using `^`. Is that intentional?

File: .prettierrc [hunk 4]
```diff
@@ -2,6 +2,9 @@
     "endOfLine": "auto",
     "tabWidth": 4,
     "printWidth": 120,
+    "trailingComma": "es5",
```
Comment (mjfwebb) [inline .prettierrc:5]: :+1:

File: package.json (hunk 3 above)
Comment (angrykoala) [reply package.json:58]: Not sure tbh, My best guess is that the pinner dependencies are the things that have been updated by changeset

**Pull request merged**